### PR TITLE
Addition of the Binepad BNR1 rotary encoder

### PR DESCRIPTION
**Pull request by vinorodrigues**
## Description

The Binepad BNR1 is a "1 knob multiple job" rotary encoder "keyboard".

BNR1 is a multifunction knob, which can be rotated, pressed, and rotated while pressed. These actions can be programmed to invoke specific responses from a range of computer applications, such as video editing, volume, brightness, media player and many more.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

*(Not applicable)*

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/binepad/bnr1/keymaps/binepad/config.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* Copyright 2022 Binepad (@binpad) */
```
Comment (zvecr) [inline keyboards/binepad/bnr1/keymaps/binepad/config.h:1]: File should be removed as it only contains VIAL defines, and those are not accepted to this repo.

File: keyboards/binepad/bnr1/keymaps/binepad/rules.mk [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+LTO_ENABLE = yes
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
+
+ENCODER_MAP_ENABLE = yes
+
+QMK_SETTINGS = no
+KEY_OVERRIDE_ENABLE = no
+COMBO_ENABLE = no
```
Comment (zvecr) [inline keyboards/binepad/bnr1/keymaps/binepad/rules.mk:12]:
```suggestion
```

File: keyboards/binepad/bnr1/keymaps/binepad/vial.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (zvecr) [inline keyboards/binepad/bnr1/keymaps/binepad/vial.json:1]: File should be removed

File: keyboards/binepad/bnr1/keymaps/default/rules.mk [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Copyright 2022 Binepad (@binpad)
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+LTO_ENABLE = yes
+VIA_ENABLE = yes
```
Comment (zvecr) [inline keyboards/binepad/bnr1/keymaps/default/rules.mk:5]: Should not be enabled on a default keymap
```suggestion
```

File: keyboards/binepad/bnr1/rules.mk [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# settings aree now in info.json
+# .. or in keymap specific rules.mk
```
Comment (zvecr) [inline keyboards/binepad/bnr1/rules.mk:2]:
```suggestion
# This file intentionally left blank
```

File: keyboards/binepad/bnr1/info.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    "processor": "atmega32u4",
+    "url": "http://binepad.com",
+    "usb": {
+        "vid": "0x4249",
+        "pid": "0x4231",
+        "device_version": "1.0.0"
+    },
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/binepad/bnr1/info.json:34]: > If you have only a single layout you should call this macro `LAYOUT`

Also, keymaps should also be refactored to reflect.
```suggestion
        "LAYOUT": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/binepad/bnr1/info.json (hunk 6 above)
Comment (vinorodrigues) [reply keyboards/binepad/bnr1/info.json:34]: ok *(didn't know if this applied to knobs, but makes no diff I suppose)*

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**